src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -62,85 +119,108 @@ int FilterParserThread::parseDATFilterFile()
         return ruleCount;
     }
 
-    unsigned int nbLine = 0;
-    while (!file.atEnd() && !m_abort) {
-        ++nbLine;
-        QByteArray line = file.readLine();
-        // Ignoring empty lines
-        line = line.trimmed();
-        if (line.isEmpty()) continue;
-        // Ignoring commented lines
-        if (line.startsWith('#') || line.startsWith("//")) continue;
+    static const int bufferSize = 2 * 1024 * 1024; // 2 MiB
+    char *buffer = new char[bufferSize];
+    qint64 bytesRead = 0;
+    int offset = 0;
+    int start = 0;
+    int endOfLine = -1;
+    int nbLine = 0;
+
+    while (true) {
+        memset(buffer + offset, 0, bufferSize - offset);
```

Comment:
I can't remember the exact reason because this is part of my previous old attempt. I think I do just to make sure there aren't garbage data after our move of the remaining data to the front of the buffer. (In case the parsing goes past the end of the read data).
I could remove it if you think this is superfluous.